domain.go

```diff
@@ -24,8 +25,29 @@ type DomainConnection struct {
 	TotalCount int      `json:"totalCount" graphql:"-"`
 }
 
-func (d *Domain) AliasOwnerType() AliasOwnerTypeEnum {
-	return AliasOwnerTypeEnumDomain
+func (d *Domain) ReconcileAliases(client *Client, aliasesWanted []string) error {
+	var allErrors, err error
+
+	aliasesToCreate, aliasesToDelete := ExtractAliases(d.ManagedAliases, aliasesWanted)
+	for _, alias := range aliasesToDelete {
+		err := client.DeleteAlias(AliasDeleteInput{
+			Alias:     alias,
+			OwnerType: AliasOwnerTypeEnumDomain,
+		})
+		allErrors = errors.Join(allErrors, err)
+	}
+
+	if len(aliasesToCreate) > 0 {
+		// CreateAliases returns current list of aliases of Domain
+		d.ManagedAliases, err = client.CreateAliases(d.Id, aliasesToCreate)
+		allErrors = errors.Join(allErrors, err)
+	} else {
+		d.ManagedAliases = slices.DeleteFunc(d.ManagedAliases, func(alias string) bool {
```

Comment:
I would think you would want to do this is regardless if len of aliasesToCreate > 0 ? 🤔

Reply:
This definitely needs a comment explaining what's going on. 
`d.ManagedAliases` is updated via two paths. 
1. In the `if len(...) > 0` case, `client.CreateAliases()` returns the latest slice of aliases from the API which is stored in `d.ManagedAliases`. No local update needed.
2. In the `else` case,  `d.ManagedAliases` is updated manually since we're done with API calls here